### PR TITLE
[Improvement-11339] Split ExecutionStatus to WorkflowExecutionStatus and TaskExecutionStatus

### LINKED ISSUE #11339: [Improvement][Master] Split ExecutionState into WorkflowExecutionState and TaskExecutionState
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Right now, we use ExecutionState to represent workflow instance status and task instance status, this make it's hard to maintain the status, since we cannot distinguish the status is belong to workflow instance or task instance. 
And the front-end can not display well.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #11339 

## Brief change log

- Split ExecutionStatus to WorkflowExecutionStatus and TaskExecutionStatus
- Remove ExecutionStatus

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@caishunfeng Please take a look, this PR is ready to review.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 1]
```diff
@@ -642,23 +638,21 @@ private Constants() {
...
-        ExecutionStatus.SUBMITTED_SUCCESS.ordinal(),
-        ExecutionStatus.DISPATCH.ordinal(),
-        ExecutionStatus.RUNNING_EXECUTION.ordinal(),
-        ExecutionStatus.DELAY_EXECUTION.ordinal(),
-        ExecutionStatus.READY_PAUSE.ordinal(),
-        ExecutionStatus.READY_STOP.ordinal(),
-        ExecutionStatus.NEED_FAULT_TOLERANCE.ordinal(),
-        ExecutionStatus.WAITING_THREAD.ordinal(),
-        ExecutionStatus.WAITING_DEPEND.ordinal()
+    public static final int[] NOT_TERMINATED_STATES = new int[]{
```
Comment (caishunfeng) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:641]: Should we split this array?

**Review by caishunfeng — APPROVED** [review]
LGTM overall

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:641]: I will fix this in other PR.

**Comment by elenore233** [comment]
@ruanwenjun Hi, I have a problem with your PR, can you help me with it?
when I try to stop a process instance with a task node(we'll say x) of SubProcess type, I found the parent process instance' state is always READY_STOP(4), never change to STOP(5), 
and the task instance of node(x)' state is RUNNING_EXECUTION(1), but the state of sub process instance (related to node(x)) is STOP(5), 

server log:
image

code:
image

can you help complete the TODO things?

**Pull request merged**